### PR TITLE
Adds doxygen docs for icalparser.h

### PR DESCRIPTION
**Pull request by xfbs**
So, this is my attempt at adding doxygen documentation for icalparser. I hope I didn't get (too much) wrong. 

There's a couple things I haven't yet managed to figure out, like on my end it shows all `icalparser_state` enum members twice for some odd reason and I can't seem to get autolinking working for all typedefs. But hey, it's a start.

What do you think?

### CODE REVIEW

**Comment by xfbs** [comment]
This would address #175

File: src/libical/icalparser.h [hunk 1]
```diff
@@ -25,60 +25,311 @@
...
+typedef struct icalparser_impl icalparser;
+
+/**
+ * @enum icalparser_state
+ * @typedef icalparser_state
+ * @brief Represents the current state of the parser
+ */
 typedef enum icalparser_state
 {
+    /** An error occured while parsing. */
```
Comment (winterz) [inline src/libical/icalparser.h]: occurred

File: src/libical/icalparser.h [hunk 2]
```diff
@@ -25,60 +25,311 @@
...
+ * @param parser The parser to use
+ * @param str A string representing a single line of RFC5545-formatted iCalendar data
+ * @return When this was the last line of the component to be parsed,
+ *  it returns the icalcomponent, otherwise it returns `NULL`.
+ * @sa icalparser_parse()
+ *
+ * @par Error handling
+ * -   If @a parser is `NULL`, it returns `NULL` and sets ::icalerrno to
+ *     ::ICAL_BADARG_ERROR.
+ * -   If @a line is empty, if retrns `NULL`
```
Comment (winterz) [inline src/libical/icalparser.h]: spelling "retrns"

File: src/libical/icalparser.h [hunk 3]
```diff
@@ -25,60 +25,311 @@
...
+ * @return The parsed icalcomponent
+ * @sa icalparser_parse_string()
+ *
+ * Reads an icalcomponent using the supplied @a line_gen_func, returning the parsed
+ * component (or `NULL` on error).
+ *
+ * @par Error handling
+ * -   If @a parser is `NULL`, it returns `NULL` and sets ::icalerrno to
+ *     ::ICAL_BADARG_ERROR.
+ * -   If data read by @a line_gen_func is empty, if retrns `NULL`
```
Comment (winterz) [inline src/libical/icalparser.h]: spelling "retrns"

**Comment by xfbs** [comment]
Merci beaucoup for the feedback — I've fixed 'em right up 👍

**Pull request merged**